src/components/AddItemForm/AddItemForm.js

```diff
@@ -0,0 +1,115 @@
+import { useState } from 'react';
+import { db } from '../../lib/firebase.js';
+
+const AddItemForm = ({ listId }) => {
+  const defaultFormValues = {
+    itemName: '',
+    purchaseInterval: '7',
+  };
+
+  const [formValues, setFormValues] = useState(defaultFormValues);
+
+  // generic function updates formValues state for any of the below form inputs
+  const handleChange = (event) => {
+    const inputName = event.target.name;
+    setFormValues({ ...formValues, [inputName]: event.target.value });
+  };
+
+  // on form submit, add user input and other default values to database
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    const newItem = formValues;
+    newItem.purchaseInterval = parseInt(newItem.purchaseInterval); // convert purchaseInterval to integer
+    newItem.lastPurchaseDate = null;
+    newItem.numberOfPurchases = 0;
+
+    try {
+      await db.collection(`lists/${listId}/items`).add(formValues); // add item to Firestore database
+      setFormValues(defaultFormValues); // after saving to database, reset form values to defaults
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <form name="addItemForm" onSubmit={handleSubmit} className="add-item-form">
+      <label
+        className="add-item-form__label add-item-form__label_type_text label"
+        htmlFor="itemName"
+      >
+        Item name:
+      </label>
+      <input
+        className="add-item-form__text-field text-field"
+        type="text"
+        id="itemName"
+        name="itemName"
+        aria-describedby="itemNameHint"
+        value={formValues.itemName}
+        onChange={handleChange}
+        maxLength="100"
+        required
```

Comment:
We added in these basic limits on the field, but I'd ideally like to do some form validation with feedback to the user. That was the intention of the empty span below, but we didn't have a chance to finish.

Reply:
This is a great instinct to have. I would gently suggest that when you come across a situation where you run out of time, create an issue to address the item that you would like to continue on. This way no one comes across this span while working on another feature which could lead to merge conflicts.

Reply:
Yes, @Cjstoney, Connie and I talked about that and opening an issue for form validation is on my to do list!